Sofa/framework/Geometry/src/sofa/geometry/Edge.h

```diff
@@ -121,6 +121,54 @@ struct Edge
     }
 
 
+    /**
+    * @brief	Test if a point is on Edge (n0, n1)
+    * @tparam   Node iterable container
+    * @tparam   T scalar
+    * @param	p0: position of the point to test
+    * @param	n0,n1: nodes of the edge
+    * @return	bool result if point is on Edge.
+    */
+    template<typename Node,
+        typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
+        typename = std::enable_if_t<std::is_scalar_v<T>>
+    >
+        static constexpr bool isPointOnEdge(const Node& p0, const Node& n0, const Node& n1)
+    {
+        static_assert(std::is_same_v < Node, sofa::type::Vec<2, T> > || std::is_same_v < Node, sofa::type::Vec<3, T> >,
```

Comment:
it is unclear to me why the 2/3 constraints are implemented with the static assert instead of the enable_if

Reply:
from my understanding to avoid using this method, at the compilation time, with type different from what is supported.